client/post-editor/editor-sharing/publicize-services.jsx

```diff
@@ -1,59 +1,66 @@
-var serviceConnections = require( 'my-sites/sharing/connections/service-connections' ),
-	EditorSharingPublicizeConnection = require( './publicize-connection' );
+import EditorSharingPublicizeConnection from './publicize-connection';
+import { getCurrentUserId } from 'state/current-user/selectors';
+import { getSelectedSiteId } from 'state/ui/selectors';
+import { getSiteUserConnections } from 'state/sharing/publicize/selectors';
 
-module.exports = React.createClass( {
-	displayName: 'EditorSharingPublicizeServices',
+class EditorSharingPublicizeServices extends Component {
+	static propTypes = {
+		post: PropTypes.object,
+		connections: PropTypes.array.isRequired,
+		newConnectionPopup: PropTypes.func.isRequired
+	};
 
-	propTypes: {
-		post: React.PropTypes.object,
-		siteId: React.PropTypes.number.isRequired,
-		connections: React.PropTypes.array.isRequired,
-		newConnectionPopup: React.PropTypes.func.isRequired
-	},
-
-	renderServices: function() {
-		var services = serviceConnections.getServicesFromConnections( this.props.connections );
-
-		return services.map( function( service ) {
-			return (
-				<li key={ service.ID } className="editor-sharing__publicize-service">
-					<h5 className="editor-sharing__publicize-service-heading">{ service.label }</h5>
-					{ this.renderConnections( service.ID ) }
-				</li>
-			);
-		}, this );
-	},
+	static defaultProps = {
+		post: Object.freeze( {} ),
+	};
 
-	renderConnections: function( serviceName ) {
-		const connections = serviceConnections.getConnectionsAvailableToCurrentUser(
-			serviceName,
-			this.props.connections
-		);
+	render() {
+		const services = reduce( this.props.connections, ( memo, connection ) => {
+			if ( connection.label in memo ) {
+				memo[ connection.label ].push( connection );
+			} else {
+				memo[ connection.label ] = [ connection ];
+			}
 
-		return connections.map( function( connection ) {
-			return (
-				<EditorSharingPublicizeConnection
-					key={ connection.ID }
-					post={ this.props.post }
-					connection={ connection }
-					onRefresh={ this.props.newConnectionPopup } />
-			);
-		}, this );
-	},
+			return memo;
+		}, {} );
 
-	render: function() {
 		return (
 			<ul className="editor-sharing__publicize-services">
-				{ this.renderServices() }
+				{ map( services, ( connections, label ) =>
+					<li key={ label } className="editor-sharing__publicize-service">
+						<h5 className="editor-sharing__publicize-service-heading">{ label }</h5>
+						{ connections.map( ( connection ) =>
+							<EditorSharingPublicizeConnection
+								key={ connection.ID }
+								post={ this.props.post }
+								connection={ connection }
+								onRefresh={ this.props.newConnectionPopup } />
+						) }
+					</li>
+				) }
 			</ul>
 		);
 	}
-} );
+}
+
+export default connect(
+	( state ) => {
+		const siteId = getSelectedSiteId( state );
+		const userId = getCurrentUserId( state );
+
+		return {
+			connections: getSiteUserConnections( state, siteId, userId ),
```

Comment:
> we could run into edge cases where we [are] thinking it would only return connections for the other user, but if getSomeOtherUserId were to return undefined, it would trigger the "current user" fallback behavior

Good point! A new selector is definitely the more sensible way